src/classes/PSSModel.ts

```diff
@@ -64,113 +68,38 @@ export default class PSSModel {
-                    task instanceof AntiTask &&
-                    task.startDate === startDate &&
-                    task.startTime === recurringTask.startTime
-                )
-              )
-                return `New transient task conflicts with an existing recurring task "${
-                  recurringTask.name
-                }" on ${getDateTime(startDate, recurringTask.startTime)}`;
-            }
-          }
-        }
-        return true;
-      case "anti":
-        // Check for existing antitask (we don't need to check duration)
-        for (const antiTask of this.tasks.filter((task): task is AntiTask => task instanceof AntiTask)) {
-          return `New antitask conflicts with an existing anti-task ${antiTask.name}.`;
-        }
-
-        // Check for valid recurring task
-        for (const recurringTask of this.tasks.filter((task): task is RecurringTask => task instanceof RecurringTask)) {
-          // Check for same day
-          if (
-            recurringTask.frequency === Frequency.Daily ||
-            (recurringTask.frequency === Frequency.Weekly &&
-              dayOfTheWeek(recurringTask.startDate) === dayOfTheWeek(recurringTask.endDate)) ||
-            (recurringTask.frequency === Frequency.Monthly &&
-              getDayOfMonth(recurringTask.startDate) === getDayOfMonth(startDate))
-          ) {
-            // Check if anti-task is within date range and correct time
-            if (
-              startDate >= recurringTask.startDate &&
-              startDate <= recurringTask.endDate &&
-              startTime === recurringTask.startTime &&
-              duration === recurringTask.duration
-            ) {
-              return true;
-            }
-          }
-        }
-        return `Unable to find recurring task that starts at ${getDateTime(
-          startDate,
-          startTime
-        )} and lasts for ${duration} minutes`;
-      case "recurring":
-        // Check for conflicting transient task
-        for (const transientTask of this.tasks.filter((task): task is TransientTask => task instanceof TransientTask)) {
-          // Check for same day
-          if (
-            Frequency.Daily ||
-            (frequency === Frequency.Weekly && dayOfTheWeek(startDate) === dayOfTheWeek(endDate ?? -1)) ||
-            (frequency === Frequency.Monthly && getDayOfMonth(transientTask.startDate) === getDayOfMonth(startDate))
-          ) {
-            // Check for overlapping time
-            if (
-              startTime < calcEndTime(transientTask.startDate, transientTask.duration) &&
-              transientTask.startDate < calcEndTime(startDate, duration)
-            ) {
-              return `New recurring tasks conflicts with existing transient task "${
-                transientTask.name
-              }" at ${getDateTime(transientTask.startDate, transientTask.startTime)}`;
-            }
-          }
-        }
+    // Check for overlapping transient tasks
```

Comment:
Were the checks for recurringtasks and antitasks intentionally removed? I believe this new code fails to check two cases:
1. anti-task must overlap with a recurring task in both startTime, duration, and date
2. every occurrence of a recurring task cannot overlap with a transient task (or another recurring task occurrence)

Reply:
Yeah you're right. Thank you for the catch